### ISSUE TITLE
Architektura w projekcie

### ISSUE THREAD

**Issue by mzastue** [Frontend-Next/.github#3]
Tak jak rozmawialiśmy offline w biurze. Proponuje potem dorzucić podejście Clean architecture do każdego repo, czy to host, czy to remote. Pewnie będzie to wymagało jakiegoś uświadomienia, ale no hej rozwijamy się, nie? ;D

Podrzucę tu potem linki jakieś i dodatkowo możemy sobie pogadać na szybko o tym później, czy to online czy offline. Co Ty na to?

**Comment by marcin-kopanski** [comment]
jasne, podrzuc linki i zrobimy o tym wiki, na przyszlosc siadziemy sobie w biurze i pogadamy co chcemy z tego uwzglednic

**Comment by mzastue** [comment]
Dobra - ogólnie myślałem o tym, żeby mieć podejście Clean Architecture i można to spróbować połączyć ze Screaming Architecture:
1. Clean Architecture - https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html
2. Screaming Architecture https://blog.cleancoder.com/uncle-bob/2011/09/30/Screaming-Architecture.html

Oby dwa linki od Wujka. Przykładowe repo:
1. https://github.com/bespoyasov/frontend-clean-architecture/tree/master
2. https://github.com/xurxodev/frontend-clean-architecture/tree/master

ℹ  Keep in mind
- Monorepo (turbo, nx, pnpm workspaces)

**Comment by marcin-kopanski** [comment]
clean mi się podoba, turborepo muszę się przyjrzeć i trochę poużywać żeby się wypowiedzieć więc dzisiaj do tego usiądę :)

**Comment by marcin-kopanski** [comment]
wpadłem za głęboko, Next JS w turborepo z mikrofronendami :D

**Comment by mzastue** [comment]
Prawda, jest głęboko, ale do ogarnięcia :D Ogólnie turborepo tylko spina wszystko z jednego ekosystemu (najlepiej nie wielkiego) w jedno repo.

Ogólnie myślę, że fajnie mieć monorepo jako bazę dla konkretnego mikrofrontu po to aby mieć jasny podział na główną aplikację (web) oraz te wszystkie rzeczy które się na nią składają. Czyli UI, logika, jakieś narzędzia, storybook itd. Nie trzeba oczywiście tego robić, wydaje mi się, że ustalałoby to fajne granice na podstawie których może da się wydzielić coś wspólnego albo inny moduł.

Ofc, można tez mieć bez monorepo, wtedy po prostu wszystko jest zbundlowane w jedną papkę :D
